From 3d9237e3856dd14f68d93bf5a1bb13e2bd82b369 Mon Sep 17 00:00:00 2001
From: Tien Nguyen <tien.nguyen.uh@renesas.com>
Date: Fri, 8 May 2026 06:46:57 +0000
Subject: [PATCH] arm64: dts: renesas: imdt-v2h-sbc: add drp-ai node

This commit adds DRP-AI node to imdt-v2h-sbc device tree

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
---
 arch/arm64/boot/dts/renesas/imdt-v2h-sbc.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/imdt-v2h-sbc.dts b/arch/arm64/boot/dts/renesas/imdt-v2h-sbc.dts
index da544bc7f999..6847aaabd0cd 100644
--- a/arch/arm64/boot/dts/renesas/imdt-v2h-sbc.dts
+++ b/arch/arm64/boot/dts/renesas/imdt-v2h-sbc.dts
@@ -76,6 +76,15 @@ drp_reserved: DRP-AI@240000000 {
 			reg = <0x2 0x40000000 0x0 0x20000000>;
 		};
 	};
+
+	udmabuf@0 {
+		compatible = "ikwzm,u-dma-buf";
+		device-name = "udmabuf0";
+		minor-number = <0>;
+		size = <0x4000000>; // 64MiB
+		dma-coherent;
+		memory-region = <&image_buf0>;
+	};
 };
 
 &sci0 {
@@ -130,3 +139,9 @@ &cru3 {
 &adc {
 	status = "okay";
 };
+
+&drpai0 {
+	memory-region = <&drp_reserved>;
+	image-memory-region = <&image_buf0>;
+	status = "okay";
+};
-- 
2.43.0

